File: template/dotnet/Agent365Adapter.cs
namespace AgnosticAgentTemplate;

public sealed class Agent365Adapter
{
    private readonly AppConfig _config;

    public Agent365Adapter(AppConfig config)
    {
        _config = config;
        // TODO: Initialize the real Agent365 SDK client with tenant/app credentials.
    }

    public Task ReportTurnStartAsync(TurnContext context)
    {
        // TODO: Map this to Agent365 telemetry/reporting API.
        Console.WriteLine($"[Agent365] Turn started: {context.TurnId}");
        return Task.CompletedTask;
    }

    public Task ReportPurviewDecisionAsync(TurnContext context, string stage, Decision decision)
    {
        // TODO: Map this to Agent365 telemetry/reporting API.
        Console.WriteLine($"[Agent365] Stage={stage}, Turn={context.TurnId}, Block={decision.Block}");
        return Task.CompletedTask;
    }

    public Task ReportTurnEndAsync(TurnContext context, ModelResult result)
    {
        // TODO: Map this to Agent365 telemetry/reporting API.
        Console.WriteLine($"[Agent365] Turn ended: {context.TurnId}, Blocked={result.Blocked}");
        return Task.CompletedTask;
    }
}

File: template/rust/src/agent365_adapter.rs
use crate::config::AppConfig;
use crate::models::{Decision, ModelResult, TurnContext};

#[derive(Clone, Debug)]
pub struct Agent365Adapter {
    #[allow(dead_code)]
    config: AppConfig,
}

impl Agent365Adapter {
    pub fn new(config: AppConfig) -> Self {
        // TODO: Initialize the real Agent365 SDK client with tenant/app credentials.
        Self { config }
    }

    pub fn report_turn_start(&self, context: &TurnContext) {
        // TODO: Map this to Agent365 telemetry/reporting API.
        println!("[Agent365] Turn started: {}", context.turn_id);
    }

    pub fn report_purview_decision(&self, context: &TurnContext, stage: &str, decision: &Decision) {
        // TODO: Map this to Agent365 telemetry/reporting API.
        println!(
            "[Agent365] Stage={}, Turn={}, Block={}",
            stage, context.turn_id, decision.block
        );
    }

    pub fn report_turn_end(&self, context: &TurnContext, result: &ModelResult) {
        // TODO: Map this to Agent365 telemetry/reporting API.
        println!(
            "[Agent365] Turn ended: {}, Blocked={}",
            context.turn_id, result.blocked
        );
    }
}
